.circleci/config.yml

```diff
@@ -58,6 +58,36 @@ commands:
           name: Ensure sudo is installed, otherwise install it.
           command: ./scripts/sudo-check.sh
 
+  upload-codecov:
+    description: Uploads testing code coverage results to Codecov
+    parameters:
+      component:
+        description: The component of the application being tested, either backend or frontend.
+        type: enum
+        enum: ["backend", "frontend"]
+      coverage-report:
+        description: The path to the coverage report being uploaded.
+        type: string
+    steps:
+      - run:
+          name: Ensure Codecov uploader is installed, otherwise install it.
+          command: ./scripts/codecov-check.sh
+      - run:
+          name: Determine Codecov metric flag
+          command: |
+            if [ "$CIRCLE_BRANCH" == "main" ] ; then
+              CURRENT_FLAG=main-<<parameters.component>>
+            elif [ "$CIRCLE_BRANCH" == "raft-tdp-main" ] ; then
```

Comment:
This logic was mostly lifted from the old setup, but I don't see a reason to not upload coverage reports on every branch. Having the Codecov information presented prior to merging a PR could definitely be very helpful, as shown [in this comment](https://github.com/raft-tech/TANF-app/pull/1149#issuecomment-887089776)

@raft-tech/tdrs-developers do any of you have strong feelings on this? The `elif` clause could just be reduced to an `else` and all branches would get these coverage reports.

Reply:
I prefer the elif because of it's readability and verbosity

Reply:
If we wanted to upload these coverage metrics for all branches the elif wouldn't need to compare anything in its clause. Specifically, I was wondering if we want to upload these metrics for all branches instead of just `main` and `raft-tdp-main`?

Reply:
Oh, I missed this larger context. Yes, this is a great improvement! It would be undesirable for everything to get through raft/qasp only to discover we fell below our codecov upon merging. I looked into the codecov functionality and it seems to not have a simple way to fail a build on too low of a coverage value. Perhaps we should have an add-on ticket for that devops automation?

Reply:
For pytest it looks like we can just pass a flag to the command we already use to enforce that - [ref](https://pytest-cov.readthedocs.io/en/latest/config.html?highlight=--cov-fail-under#reference)

For the frontend it looks like we already should fail if we go under 90%. From package.json:
```
"test:ci": "CI=1 react-scripts test --coverage",
....
"coverageThreshold": {
      "global": {
        "statements": 90,
        "branches": 90,
        "functions": 90,
        "lines": 90
      }
    }
```